src/Workspaces/Core/Portable/Storage/SQLite/SQLitePersistentStorage.Accessor.cs

```diff
@@ -135,21 +120,24 @@ public Task<Stream> ReadStreamAsync(TKey key, Checksum checksum, CancellationTok
                         var (checksumBytes, checksumLength, checksumPooled) = GetBytes(checksumOpt, cancellationToken);
                         var (dataBytes, dataLength, dataPooled) = GetBytes(stream);
 
-                        await AddWriteTaskAsync(key, con =>
+                        using (var pooledConnection = Storage.GetPooledConnection())
```

Comment:
note: we are no longer asynchronous.  so we could consider just extending the range of the higher up pooled-connection `using` to just encompass this.